tests/test_iqm_provider.py

```diff
@@ -312,6 +354,31 @@ def test_run_with_heralding_mode_zeros(backend, circuit, submit_circuits_default
     backend.run([circuit], heralding_mode='zeros')
 
 
+def test_run_with_circuit_callback(backend, job_id, submit_circuits_default_kwargs):
+    qc1 = QuantumCircuit(3)
+    qc1.measure_all()
+    qc2 = QuantumCircuit(3)
+    qc2.r(np.pi, 0.3, 0)
+    qc2.measure_all()
+
+    def sample_callback(circuits) -> None:
```

Comment:
You could add a check that the callback is actually called, e.g. by using some mock and asserting calls.

Reply:
Does it look good enough now?